packages/@aws-cdk/aws-redshift-alpha/test/cluster-parameter-change-reboot-handler.test.ts

```diff
@@ -0,0 +1,367 @@
+              ParameterApplyStatus: 'retry',
+            }],
+          }],
+        };
+      }).mockImplementationOnce(() => {
+        return {
+          Clusters: [{
+            ClusterParameterGroups: [{
+              ParameterGroupName: 'parameter-group-name',
+              ParameterApplyStatus: 'pending-reboot',
+            }],
+          }],
+        };
+      });
+
+    const event: Partial<AWSLambda.CloudFormationCustomResourceEvent> = {
+      RequestType: 'Create',
+      ResourceProperties: {
+        ServiceToken: 'service-token',
+        ClusterId: 'cluster-id',
+        ParameterGroupName: 'parameter-group-name',
+      },
+    };
+
+    // WHEN
+    await invokeHandler(event);
+
+    // THEN
+    expect(setTimeout).toHaveBeenCalled();
+    expect(mockDescribeClusters).toHaveBeenCalledTimes(2);
+    expect(mockRebootCluster).toHaveBeenCalledTimes(1);
+  });
+
+  test('retries if rebootCluster throws InvalidClusterStateFault error', async () => {
+    // GIVEN
+    mockDescribeClusters.mockImplementation(() => {
+      return {
+        Clusters: [{
+          ClusterParameterGroups: [{
+            ParameterGroupName: 'parameter-group-name',
+            ParameterApplyStatus: 'pending-reboot',
+          }],
+        }],
+      };
+    });
+    mockRebootCluster
+      .mockImplementationOnce(async () => {
+        const { InvalidClusterStateFault } = jest.requireActual('@aws-sdk/client-redshift');
+        return Promise.reject(new InvalidClusterStateFault());
+      })
+      .mockImplementationOnce(jest.fn());
+
+    const event: Partial<AWSLambda.CloudFormationCustomResourceEvent> = {
+      RequestType: 'Create',
+      ResourceProperties: {
+        ServiceToken: 'service-token',
+        ClusterId: 'cluster-id',
+        ParameterGroupName: 'parameter-group-name',
+      },
+    };
+
+    // WHEN
+    await invokeHandler(event);
+
+    // THEN
+    expect(setTimeout).toHaveBeenCalled();
+    expect(mockDescribeClusters).toHaveBeenCalledTimes(1);
+    expect(mockRebootCluster).toHaveBeenCalledTimes(2);
+  });
+
+  test('retries if rebootCluster throws generic error', async () => {
+    // GIVEN
+    mockDescribeClusters.mockImplementation(() => {
+      return {
+        Clusters: [{
+          ClusterParameterGroups: [{
+            ParameterGroupName: 'parameter-group-name',
+            ParameterApplyStatus: 'pending-reboot',
+          }],
+        }],
+      };
+    });
+    mockRebootCluster.mockImplementation(async () => {
+      return Promise.reject(new Error('error'));
+    });
+
+    const event: Partial<AWSLambda.CloudFormationCustomResourceEvent> = {
+      RequestType: 'Create',
+      ResourceProperties: {
+        ServiceToken: 'service-token',
+        ClusterId: 'cluster-id',
+        ParameterGroupName: 'parameter-group-name',
+      },
+    };
+
+    // WHEN
+    // THEN
+    await expect(() => invokeHandler(event)).rejects.toThrow('error');
+  });
```

Comment:
Could you elaborate on this test? How is this retrying? It looks like it is just failing.

Reply:
You're right, I gave the wrong description.
Thanks for the spot 💪